src/chttpd/src/chttpd_db.erl

```diff
@@ -1631,6 +1626,17 @@ db_attachment_req(#httpd{method=Method, user_ctx=Ctx}=Req, Db, DocId, FileNamePa
     end,
 
     #doc{atts=Atts} = Doc,
+    case Method of
+        'DELETE' ->
+            case get_attachment(Req, Db, DocId, FileName) of
```

Comment:
Since we already did a fabric:open_revs calls to get the document, we could try to save another fabric round-trip by just checking that the Atts is has `FileName`

```
+    case Method of
+        'DELETE' ->
+            case [A || A <- Atts, couch_att:fetch(name, A) == FileName] of
+                [] ->
+                     throw({not_found, "Document is missing attachment"});
+                _ ->
+                    ok
+            end
+    end,
```